### PR TITLE
[top_earlgrey/rtl] adding csrng and edn blocks

### PR DESCRIPTION
**Pull request by mwbranstad**
Most connections for entropy complex blocks add been made.

Signed-off-by: Mark Branstad <mark.branstad@wdc.com>

### CODE REVIEW

File: hw/ip/csrng/csrng.core
Comment (msfschaffner) [inline hw/ip/csrng/csrng.core:12]: I believe this should be `:prim:multibit_sync`...

**Review by senelson7 — CHANGES_REQUESTED** [review]
In edn.hjson file I'm assuming the "resval" field for the CMD_RDY and CMD_STS bits in SW_CMD_STS register stands for "reset value" and should both be "0".

**Comment by senelson7** [comment]
Mark indicated that the cmd_rdy bit really should power up to "0" and it's then controlled be an edn input so my previous comment/request should be ignored.

**Review by senelson7 — APPROVED** [review]

**Comment by mwbranstad** [comment]
@eunchan @msfschaffner can you check the inter signal connections between csrng and edn. I was unable to connect more than one port of the csrng app interface because the DV ENV does not appear to support multiple reg instances of a module, at least out of the box. Because of this, lots of TODOs were left until support is available.
@sriyerg can you confirm that my observation is correct, and what the fix might be.

**Comment by weicaiyang** [comment]
> @eunchan @msfschaffner can you check the inter signal connections between csrng and edn. I was unable to connect more than one port of the csrng app interface because the DV ENV does not appear to support multiple reg instances of a module, at least out of the box. Because of this, lots of TODOs were left until support is available.
> @sriyerg can you confirm that my observation is correct, and what the fix might be.

If it's just internal signal increased, I feel DV ENV should support it. Let me try this out.

**Comment by weicaiyang** [comment]
> > @eunchan @msfschaffner can you check the inter signal connections between csrng and edn. I was unable to connect more than one port of the csrng app interface because the DV ENV does not appear to support multiple reg instances of a module, at least out of the box. Because of this, lots of TODOs were left until support is available.
> > @sriyerg can you confirm that my observation is correct, and what the fix might be.
> 
> If it's just internal signal increased, I feel DV ENV should support it. Let me try this out.

one failure is because we don't support generate RAL reg package for a IP with multiple instances in the top. Let me see if I can quickly add that support.

**Comment by weicaiyang** [comment]
> > > @eunchan @msfschaffner can you check the inter signal connections between csrng and edn. I was unable to connect more than one port of the csrng app interface because the DV ENV does not appear to support multiple reg instances of a module, at least out of the box. Because of this, lots of TODOs were left until support is available.
> > > @sriyerg can you confirm that my observation is correct, and what the fix might be.
> > 
> > 
> > If it's just internal signal increased, I feel DV ENV should support it. Let me try this out.
> 
> one failure is because we don't support generate RAL reg package for a IP with multiple instances in the top. Let me see if I can quickly add that support.

just created $4309 to address this. I uncommented those TODOs in my local and Smoke is passing with that fix

**Comment by rasmus-madsen** [comment]
@mwbranstad it seems you are changing file permissions - not sure how or why or if it even matters.
any input @rswarbrick @msfschaffner @imphil 
if so I think we need a hint on how to fix this?

**Comment by rswarbrick** [comment]
Yes, it matters (other checkouts will end up with executable `.sv` files, which is probably not what we want!). Since I don't think you've touched any files that should be executable, you can do something like this to undo things:
```
git diff --name-only HEAD^ | xargs chmod 0644
```

**Comment by senelson7** [comment]
The resval change is needed for dv for 100% of the edn CSR tests to pass. Is there anything still holding up the merge of this PR? Thanks.

**Comment by weicaiyang** [comment]
> @eunchan @msfschaffner can you check the inter signal connections between csrng and edn. I was unable to connect more than one port of the csrng app interface because the DV ENV does not appear to support multiple reg instances of a module, at least out of the box. Because of this, lots of TODOs were left until support is available.
> @sriyerg can you confirm that my observation is correct, and what the fix might be.

@mwbranstad I have added this support in #4309. The PR is merged. You can uncomment out those 2 instance codes now.

**Comment by mwbranstad** [comment]
Because PR #4309 was merged, picked this up so that multiple edn blocks can now be connected.

**Review by msfschaffner — COMMENTED** [review]
Thanks @mwbranstad. This looks good overall, just a quick question regarding the memory map.

File: hw/top_earlgrey/ip/xbar_main/data/autogen/xbar_main.gen.hjson
Comment (msfschaffner) [reply hw/top_earlgrey/ip/xbar_main/data/autogen/xbar_main.gen.hjson:350]: CC @eunchan

Comment (mwbranstad) [reply hw/top_earlgrey/ip/xbar_main/data/autogen/xbar_main.gen.hjson:350]: The last time I looked at a memory map, there was not enough space open to add these addresses without overlapping an existing block. If this has changed, please provide a pointer to the new map. It may be that the memory map still needs changing, but not sure what the process is to do that.

**Review by msfschaffner — APPROVED** [review]
This LGTM now. Will merge once CI passes.

**Pull request merged**
